drivers/dma/dmamux_stm32.c

```diff
@@ -186,25 +179,28 @@ static const struct dma_driver_api dma_funcs = {
 	.stop		 = dmamux_stm32_stop,
 };
 
-#define DMAMUX_INIT(index) \
-								\
-const struct dmamux_stm32_config dmamux_stm32_config_##index = {\
-	.pclken = { .bus = DT_INST_CLOCKS_CELL(index, bus),	\
-		    .enr = DT_INST_CLOCKS_CELL(index, bits) },	\
-	.base = DT_INST_REG_ADDR(index),			\
-	.channel_nb = DT_INST_PROP(index, dma_channels),	\
-	.gen_nb = DT_INST_PROP(index, dma_generators),		\
-	.req_nb = DT_INST_PROP(index, dma_requests),		\
-};								\
-								\
-static struct dmamux_stm32_data dmamux_stm32_data_##index = {	\
-	.mux_channels = NULL,					\
-};								\
-								\
-DEVICE_AND_API_INIT(dmamux_##index, DT_INST_LABEL(index),	\
-		    &dmamux_stm32_init,				\
+#define DMAMUX_INIT(index) 						\
+									\
+const struct dmamux_stm32_config dmamux_stm32_config_##index = {	\
+	.pclken = { .bus = DT_INST_CLOCKS_CELL(index, bus),		\
+		    .enr = DT_INST_CLOCKS_CELL(index, bits) },		\
+	.base = DT_INST_REG_ADDR(index),				\
+	.channel_nb = DT_INST_PROP(index, dma_channels),		\
+	.gen_nb = DT_INST_PROP(index, dma_generators),			\
+	.req_nb = DT_INST_PROP(index, dma_requests),			\
+};									\
+									\
+static struct dmamux_stm32_channel					\
+	dmamux_stm32_chan_##index[DT_INST_PROP(index, dma_channels)];	\
+									\
+static struct dmamux_stm32_data dmamux_stm32_data_##index = {		\
+	.mux_channels = dmamux_stm32_chan_##index,			\
+};									\
```

Comment:
I'm ok with the current name as for the channel array variable, but I would have put it in plural form: 
```
dmamux_stm32_channels_##index
```

Reply:
As always naming things it the hardest part of programming :-)